zmq_addon.hpp

```diff
@@ -408,190 +408,115 @@ class multipart_t
-        assert(multipart.size() == 7);
-        assert(num == 1.0f);
-
-        str = multipart.popstr();
-        assert(multipart.size() == 6);
-        assert(str == "Frame2");
-
-        str = multipart.popstr();
-        assert(multipart.size() == 5);
-        assert(str == "Frame3");
-
-        str = multipart.popstr();
-        assert(multipart.size() == 4);
-        assert(str == "Frame4");
-
-        str = multipart.popstr();
-        assert(multipart.size() == 3);
-        assert(str == "Frame5");
-
-        str = multipart.popstr();
-        assert(multipart.size() == 2);
-        assert(str == "Frame6");
-
-        str = multipart.popstr();
-        assert(multipart.size() == 1);
-        assert(str == "Frame7");
-
-        num = multipart.poptyp<float>();
-        assert(multipart.empty());
-        assert(num == 8.0f);
-
-        // Test message constructors and concatenation
-        multipart_t head("One", 3);
-        head.addstr("Two");
-        assert(head.size() == 2);
-
-        multipart_t tail("One-hundred");
-        tail.pushstr("Ninety-nine");
-        assert(tail.size() == 2);
-
-        multipart_t tmp(message_t("Fifty", 5));
-        assert(tmp.size() == 1);
-
-        multipart_t mid = multipart_t::create(49.0f);
-        mid.append(std::move(tmp));
-        assert(mid.size() == 2);
-        assert(tmp.empty());
-
-        multipart_t merged(std::move(mid));
-        merged.prepend(std::move(head));
-        merged.append(std::move(tail));
-        assert(merged.size() == 6);
-        assert(head.empty());
-        assert(tail.empty());
-
-        ok = merged.send(output);
-        assert(merged.empty());
-        assert(ok);
-
-        multipart_t received(input);
-        assert(received.size() == 6);
-
-        str = received.popstr();
-        assert(received.size() == 5);
-        assert(str == "One");
-
-        str = received.popstr();
-        assert(received.size() == 4);
-        assert(str == "Two");
-
-        num = received.poptyp<float>();
-        assert(received.size() == 3);
-        assert(num == 49.0f);
-
-        str = received.popstr();
-        assert(received.size() == 2);
-        assert(str == "Fifty");
-
-        str = received.popstr();
-        assert(received.size() == 1);
-        assert(str == "Ninety-nine");
-
-        str = received.popstr();
-        assert(received.empty());
-        assert(str == "One-hundred");
-
-        return 0;
-    }
-
 private:
     // Disable implicit copying (moving is more efficient)
     multipart_t(const multipart_t& other) ZMQ_DELETED_FUNCTION;
     void operator=(const multipart_t& other) ZMQ_DELETED_FUNCTION;
 };  // class multipart_t
 
-#endif  // ZMQ_HAS_RVALUE_REFS
-
 inline std::ostream& operator<<(std::ostream& os, const multipart_t& msg)
 {
     return os << msg.str();
 }
 
+#endif  // ZMQ_HAS_RVALUE_REFS
+
+#if defined(ZMQ_BUILD_DRAFT_API) && defined(ZMQ_CPP11) && defined(ZMQ_HAVE_POLLER)
+    class active_poller_t
```

Comment:
Much cleaner now after moving this to separate class. Can we change a name though, please? Something more zmq'ish and shorter? Maybe loop_t, or service_t? I have some pending additions - timers - and it would make sense to put it here. Maybe with can split this actually into separate file? zmq_monitor.hpp, zmq_service.hpp?

Reply:
I am not completely satisfied with my name as well. I think "poller" should be part of the name, but "active" is not so specific. loop_t would be strange IMO, as there is no loop within wait. A "service" is also something quite different for me, which has its own thread etc. These concepts could be built on top of active_poller_t again. 

About splitting up the headers... I am somewhat undecided about that. It is a feature of cppzmq that it consists only of two header files. But on the other hand, when you can copy two header files, you can also copy a handful. If it is done, zmq_addon.hpp should include all the extra headers then.

Reply:
yeah, `naming` is the most difficult part in software development. maybe zpoller_t (z - zuper) :)
We can stay with two header for now. If have more stuff we can split then and I like an idea of zmq_addon.hpp to have all , probably with a rename to zmq_addons.hpp